ls1intum/Artemis
# Remove duplicated code in assessment service

There was quite a lot of duplicated code. This PR tries to unify it and to remove unnecessary `@Transactional` annotations.

### Checklist

### Description
In this PR we tried to reduce duplicated code and unify some steps of the exercise assessments. There should be no functional changes in this PR.

### Steps for Testing
_Some of these steps are more easily tested in a group (multiple submissions)._

Text Exercise:
1. Create text exercise with assessment suggestions.
2. Create multiple submissions (at least 5).
3. Assess the submissions (save, submit, override). 
4. Verify that the automatic assessment suggestions are working. 
5. Complain about the assessment.
6. Respond to the assessment.

Programming Exercise:
1. Create programming exercise.
2. Add submission.
3. Assess submission (save, submit, override).
4. Complain about those assessments.
5. Respond to the complaints.

Modeling Exercise:
1. Create modeling exercise with assessment suggestions.
2. Create multiple submissions (at least 5).
3. Assess the submissions (save, submit, override). 
4. Verify that the automatic assessment suggestions are working. 
5. Complain about the assessment.
6. Respond to the assessment.

### Test Coverage
ProgrammingAssessmentResource 87%
TextAssessmentResource 92%
ModelingAssessmentResource 96%
ModelingSubmissionResource 97%
FileUploadAssessmentResource 98%
FileUploadSubmissionResource 91%
AssessmentResource 93%
SubmissionResource 90%

AssessmentService 87%
ProgrammingAssessmentService 91%
TextAssessmentService 93%
ComplaintService 87%
FileUploadSubmissionService 82%
TextSubmissionService 80%
ModelingSubmissionService 80%
ParticipationService 78%
